esphome/components/airthings_ble/airthings_listener.cpp

```diff
@@ -0,0 +1,30 @@
+#include "airthings_listener.h"
+#include "esphome/core/log.h"
+
+#ifdef ARDUINO_ARCH_ESP32
+
+namespace esphome {
+namespace airthings_ble {
+
+static const char *TAG = "airthings_ble";
+
+bool AirthingsListener::parse_device(const esp32_ble_tracker::ESPBTDevice &device) {
+  for (auto &it : device.get_manufacturer_datas()) {
+    if (it.uuid == esp32_ble_tracker::ESPBTUUID::from_uint16(0x0334)) {
+      uint SN = it.data[0];
+      SN |= ((uint) it.data[1] << 8);
+      SN |= ((uint) it.data[2] << 16);
+      SN |= ((uint) it.data[3] << 24);
```

Comment:
```suggestion
      if (it.data.size() < 4)
        continue;
      uint32_t sn = it.data[0];
      sn |= ((uint32_t) it.data[1] << 8);
      sn |= ((uint32_t) it.data[2] << 16);
      sn |= ((uint32_t) it.data[3] << 24);
```

Just adds a check to prevent out-of-bounds reading, and replaces `uint` with the more concrete `uint32_t` here, and fixes variable naming to match the repos settings.